src/validation.cpp

```diff
@@ -3663,6 +3673,11 @@ void BlockManager::FindFilesToPrune(std::set<int>& setFilesToPrune, uint64_t nPr
            nLastBlockWeCanPrune, count);
 }
 
+void BlockManager::UpdatePruneBlocker(std::string name, const CBlockIndex* block) {
+    AssertLockHeld(::cs_main);
+    m_prune_blockers[name] = block;
```

Comment:
b5a239390fd1b31c7e6107a2e6c35658a2b4697c

Should check/assert it only goes forward when updating?

Reply:
I am not sure. The way I use it right now it can also go down with reorgs. I am undecided what the right course is. Only going up avoids a potential footgun, with going down it could react to (highly unlikely) deep reorg. If I add this I think I would do it so that setting a lower number is simply a no op:

```
if (m_prune_blockers[name] < block) m_prune_blockers[name] = block;
```

Reply:
But what should happen if the block is already pruned?

Reply:
Hm, that would mean the node would have to restart with `-reindex` anyway, right?

Reply:
Yes, I added functional tests for the -reindex case.

I am looking into the sanity check assert idea don't have a clear opinion yet.